app/src/main/java/de/danoeh/antennapod/fragment/FeedSettingsFragment.java

```diff
@@ -159,6 +161,37 @@ public void onDestroy() {
             }
         }
 
+        private void setupFeedAutoSkipPreference() {
+            findPreference("skipping").setOnPreferenceClickListener(preference -> {
+                new FeedPreferenceSkipDialog(getContext(),
+                        feedPreferences.getFeedSkipIntro(),
+                        feedPreferences.getFeedSkipEnding()) {
+                    @Override
+                    protected void onConfirmed(String skipIntroStr, String skipEndingStr) {
+
+                        int skipIntro;
+                        int skipEnding;
+                        try {
+                            skipIntro = Integer.parseInt(skipIntroStr);
+                        } catch (NumberFormatException e) {
+                            skipIntro = 0;
+                        }
```

Comment:
I think this could be done inside `FeedPreferenceSkipDialog` to keep the `FeedSettingsFragment` a bit shorter

Reply:
done